bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -56,39 +58,81 @@ export class StringFormatter extends CellFormatter {
   }
 
   static {
-    this.define<StringFormatter.Props>(({Color, Nullable, String}) => ({
-      font_style: [ FontStyle, "normal" ],
-      text_align: [ TextAlign, "left"   ],
-      text_color: [ Nullable(Color), null ],
+    this.define<StringFormatter.Props>(({String}) => ({
+      font_style: [ p.FontStyleSpec, {value: "normal"} ],
+      text_align: [ p.TextAlignSpec, {value: "left"} ],
+      text_color: [ p.ColorSpec, null ],
+      background_color: [ p.ColorSpec, null ],
       nan_format: [ String, "-"],
     }))
   }
 
   override doFormat(_row: any, _cell: any, value: any, _columnDef: any, _dataContext: any): string {
-    const {font_style, text_align, text_color} = this
+    const {font_style, text_align, text_color, background_color} = this
 
     const text = div(value == null ? "" : `${value}`)
-    switch (font_style) {
+
+    // Font style
+    let resolvedFontStyle
+    if (isValue(font_style)) {
+      resolvedFontStyle = font_style.value
+    } else if (isField(font_style)) {
+      resolvedFontStyle = _dataContext[font_style.field]
+    }
+
+    switch (resolvedFontStyle) {
```

Comment:
Our APIs use snake case, so:
```suggestion
    switch (resolved_font_style) {
```
and in all other cases.